helper/max_length_slice.go

```diff
@@ -0,0 +1,51 @@
+package helper
+
+import "sync"
+
+type MaxLengthStringSliceAdaptor struct {
+	s      []string
+	maxlen int
+	lock   sync.Mutex
```

Comment:
Use a `RWMutex` here to allow reading in parallel or use a channel to sync the writing actions.